internal/pkg/template/diff/diff.go

```diff
@@ -90,7 +91,75 @@ func isYAMLLeaf(node *yaml.Node) bool {
 }
 
 func parseSequence(from, to *yaml.Node) (map[string]*Node, error) {
-	return nil, nil
+	fromSeq, toSeq := make([]yaml.Node, len(from.Content)), make([]yaml.Node, len(to.Content))
+	if err := to.Decode(&toSeq); err != nil {
+		return nil, err
+	}
+	if err := from.Decode(&fromSeq); err != nil {
+		return nil, err
+	}
+	cachedDiff, cachedErr := make(map[string]*Node), make(map[string]error)
+	lcsIdxInFrom, lcsIdxInTo := longestCommonSubsequence(fromSeq, toSeq, func(inA, inB int) bool {
+		diff, err := parse(&(fromSeq[inA]), &(toSeq[inB]), "")
+		if diff != nil { // NOTE: cache the diff only if a modification could have happened at this position.
+			cachedDiff[cachedKey(inA, inB)] = diff
+			cachedErr[cachedKey(inA, inB)] = err
+		}
+		return err == nil && diff == nil // NOTE: we swallow the error for now.
```

Comment:
It's not! I was thinking "for now" as "at this point of execution" 😂  because if `err != nil`, it's most likely an un/marshaling error, and it's gonna surface sooner or later. This `eq` func just doesn't bother handling it!